app/experimenter/nimbus-ui/src/components/AppLayoutWithSidebar/index.test.tsx

```diff
@@ -108,5 +110,37 @@ describe("AppLayoutWithSidebar", () => {
       expect(overviewLink!).toHaveClass("text-dark");
       expect(overviewLink!).not.toHaveClass("text-primary");
     });
+
+    it("renders information about missing experiment details", async () => {
+      const review = {
+        ready: false,
+        missingFields: {
+          Overview: [],
+          Branches: [],
+          Metrics: [],
+          Audience: [],
+        },
```

Comment:
Why do you prefer the page names to be capitalized here?

Also, do we need to store the name of fields that are missing in these arrays or can these just be boolean? Is that to better set us up for #3760?

Reply:
> Why do you prefer the page names to be capitalized here?

I'm using these as page names when rendering the "Missing details in [page]" links. You'll see in some areas that I'm `toLowerCase()`ing them as well for usage in URLs and tests.

It could be argued that we could be more explicit here with something like:

```
[
  {
    name: "Overview",
    slug: "overview",
    missingFields: ["foo", "bar"],
  },
  {
    name: "Branches",
    slug: "branches",
    missingFields: ["baz"],
  }
]
```

But we're getting the same result either way. What do you think?

> do we need to store the name of fields that are missing in these arrays 

Yep, I have this mapped out for a future PR:

```tsx
const fieldMap = {
  Overview: ["public_description"],
  Branches: ["reference_branch"],
  Audience: [
    "channels",
    "firefox_min_version",
    "targeting_config_slug",
    "proposed_enrollment",
    "proposed_duration"
  ]
}
```

I was going to expand on this in [EXP-464](https://jira.mozilla.com/browse/EXP-464).

Reply:
> I'm using these as page names when rendering the "Missing details in [page]" links. You'll see in some areas that I'm `toLowerCase()`ing them as well for usage in URLs and tests.

Wouldn't it be easier/more consistent with (I think all of?) our other objects to lowercase them so you don't have to lowercase them in tests, then capitalize in that one instance?

> I was going to expand on this in EXP-464.

I totally see the need for `fieldMap` to check that against what's returned from the full experiment validation, but do we need to store the fields in `missingDetails` or can that be set to `true`/`false` since the sidebar doesn't care _which_ fields are missing?

Reply:
See my comment below